src/rpc/server/basic.zig

```diff
@@ -36,14 +38,30 @@ pub fn acceptAndServeConnection(server_ctx: *server.Context) !void {
     defer server_ctx.allocator.free(buffer);
 
     var http_server = std.http.Server.init(conn, buffer);
-    var request = try http_server.receiveHead();
+    var request = http_server.receiveHead() catch |err| {
+        logger.err()
+            .field("conn", conn.address)
+            .logf("Receive head error: {s}", .{@errorName(err)});
+        return;
+    };
+    const head_info = requests.HeadInfo.parseFromStdHead(request.head) catch |err| {
+        switch (err) {
+            error.RequestTargetTooLong => logger.err().field("conn", conn.address).logf(
+                "Request target was too long: '{}'",
+                .{std.zig.fmtEscapes(request.head.target)},
+            ),
+            else => |e| logger.err().field("conn", conn.address)
+                .logf("Request error: {s}", .{@errorName(e)}),
+        }
+        return;
+    };
 
-    const conn_address = request.server.connection.address;
-    logger.info().logf("Responding to request from {}: {} {s}", .{
-        conn_address, requests.methodFmt(request.head.method), request.head.target,
-    });
+    logger.info().field("conn", conn.address).logf(
+        "Responding to request: {} {s}",
+        .{ requests.httpMethodFmt(request.head.method), request.head.target },
+    );
```

Comment:
nit: debug level feels more appropriate than info for this?

Reply:
2472321624179c4c33720c672306e62190107b1c